sudo/lib/exec/parent.rs

```diff
@@ -212,15 +216,41 @@ impl ParentClosure {
             }
         }
     }
+
+    /// Handle changes to the monitor status.
+    fn handle_sigchld(&mut self) {
+        const OPTS: WaitOptions = WaitOptions::new().all().untraced().no_hang();
+
+        let (pid, status) = loop {
+            match waitpid(WaitPid::any(), OPTS) {
```

Comment:
not having too clear the parent/child relationship between the original sudo process, the PTY monitor/relay and the exec'd command, could this result in a different process (not the monitor) being returned here?

given the 
> // FIXME: what other child could we have?

question below. perhaps this loop could be guarded by an `if let Some(monitor_pid) = self.monitor_pid` and `waitpid` should use `monitor_pid` as its argument. that this, this function should only check if the monitor_pid child has finished until there's a need for executing logic when a different child returns / stops?

Reply:
mmm... the parent sudo process (aka the original sudo process) forks to create the monitor and eventually the monitor forks to create the command.

There shouldn't be any reason for us to `waitpid` from the parent and receive a status update for other process than the monitor. The only case that comes to mind is if we implement interception where the parent process forks to create other processes.

We could do the `if let Some(monitor_pid)` thing but we already do that in the `on_signal` function so I didn't want to do it twice. Maybe `handle_sigchld` should receive the `monitor_pid` as a parameter instead idk.